Libraries/LibWeb/HTML/HTMLTitleElement.cpp

```diff
@@ -50,4 +59,40 @@ void HTMLTitleElement::set_text(String const& value)
     string_replace_all(value);
 }
 
+void HTMLTitleElement::consider_propagate_title_update()
+{
+    if (text().is_empty()) {
+        m_empty_titles_in_a_row++;
+    } else {
+        m_empty_titles_in_a_row = 0;
+    }
+
+    if (m_first_block_at_ms == 0) {
+        m_first_block_at_ms = MonotonicTime::now_coarse().milliseconds();
+    }
+
+    if (MonotonicTime::now_coarse().milliseconds() - m_first_block_at_ms > timer_unconditional_update_ms) {
+        if (m_empty_titles_in_a_row == 1) {
+            // skip this update anyways. This is the first update, which should be followed by the "actual" update.
+            return;
+        }
+
+        // we've exceeded the max ms without a title update, we'll propagate it immediately
+        propagate_title_update();
+        return;
+    }
+
+    // start a throttling timer. This is to not spam every update to the front-end.
+    // furthermore, the first update is removing the title, so it gives us an erroneous empty title.
+    m_throttle_update_timer->set_active(true);
+    m_throttle_update_timer->restart();
```

Comment:
`->set_active(true)` calls `->start()`, and `->restart()` then calls `->stop()` followed by `->start()`. Can't this just be `->restart()` instead?